packages/sources/dlc-btc-por/src/transport/proof-of-reserves.ts

```diff
@@ -0,0 +1,219 @@
+import { TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { ResponseCache } from '@chainlink/external-adapter-framework/cache/response'
+import { AdapterResponse, makeLogger, sleep } from '@chainlink/external-adapter-framework/util'
+import { SubscriptionTransport } from '@chainlink/external-adapter-framework/transports/abstract/subscription'
+import { EndpointContext } from '@chainlink/external-adapter-framework/adapter'
+import { Requester } from '@chainlink/external-adapter-framework/util/requester'
+import { ethers } from 'ethers'
+import { hex } from '@scure/base'
+import { p2tr, p2tr_ns, taprootTweakPubkey, TAPROOT_UNSPENDABLE_KEY } from '@scure/btc-signer'
+import { BaseEndpointTypes } from '../endpoint/proof-of-reserves'
+import abi from '../config/dlc-manager-abi.json'
+import {
+  BitcoinNetwork,
+  BitcoinTransaction,
+  BitcoinTransactionVectorOutput,
+  getBitcoinNetwork,
+  RawVault,
+} from './utils'
+
+const logger = makeLogger('dlcBTC PoR')
+
+export type TransportTypes = BaseEndpointTypes
+
+export class DLCBTCPorTransport extends SubscriptionTransport<TransportTypes> {
+  name!: string
+  responseCache!: ResponseCache<TransportTypes>
+  requester!: Requester
+  provider!: ethers.providers.JsonRpcProvider
+  dlcManagerContract!: ethers.Contract
+  settings!: TransportTypes['Settings']
+
+  async initialize(
+    dependencies: TransportDependencies<TransportTypes>,
+    adapterSettings: TransportTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+    const { RPC_URL, CHAIN_ID, DLC_CONTRACT } = adapterSettings
+    this.settings = adapterSettings
+    this.provider = new ethers.providers.JsonRpcProvider(RPC_URL, CHAIN_ID)
+    this.dlcManagerContract = new ethers.Contract(DLC_CONTRACT, abi, this.provider)
+    this.requester = dependencies.requester
+  }
+  async backgroundHandler(context: EndpointContext<TransportTypes>) {
+    await this.handleRequest()
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  async handleRequest() {
+    let response: AdapterResponse<TransportTypes['Response']>
+    try {
+      response = await this._handleRequest()
+    } catch (e) {
+      const errorMessage = e instanceof Error ? e.message : 'Unknown error occurred'
+      logger.error(e, errorMessage)
+      response = {
+        statusCode: 502,
+        errorMessage,
+        timestamps: {
+          providerDataRequestedUnixMs: 0,
+          providerDataReceivedUnixMs: 0,
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+    }
+    await this.responseCache.write(this.name, [{ params: [], response }])
+  }
+
+  async _handleRequest(): Promise<AdapterResponse<TransportTypes['Response']>> {
```

Comment:
maybe not necessary, but should we be using `calculateHttpRequestKey` somewhere here?

Reply:
It can be used in `requester.request` to create a unique request key to coalesce HTTP requests. Since we have one endpoint/transport with no input params, the `txid` is used as a unique identifier, so i would say we don't actually need this.